mlflow/pyfunc/__init__.py

```diff
@@ -1810,10 +1994,37 @@ def spark_udf(
         # this case all executors and driver share the same filesystem
         is_spark_in_local_mode = spark.conf.get("spark.master").startswith("local")
 
+    is_dbconnect_mode = is_databricks_connect(spark)
+    if prebuilt_env_path is not None and not is_dbconnect_mode:
+        raise RuntimeError(
+            "'prebuilt_env' parameter can only be used in Databricks Serverless "
+            "notebook REPL, atabricks Shared cluster notebook REPL, and Databricks Connect client "
+            "environment."
+        )
+
+    if (
+        prebuilt_env_path is None
+        and env_manager != _EnvManager.LOCAL
+        and is_dbconnect_mode
+        and not is_in_databricks_runtime()
+    ):
+        raise RuntimeError(
+            "If using Databricks Connect to connect to Databricks cluster from your "
+            "own machine, and 'env_manager' is set to 'virtualenv', "
+            "the 'prebuilt_env_path' param is required."
+        )
```

Comment:
```suggestion
        raise RuntimeError(
            "'prebuilt_env_path' param is required if ..."
        )
```

can we say `prebuilt_env_path' param is required` first?